### ISSUE TITLE
A bug of htmlparse

### ISSUE THREAD

**Issue by zyf0330** [kangax/html-minifier#507]
I am using version 1.1.1, and there is a bug. I have tested with 1.2.0 version in http://kangax.github.io/html-minifier/ , the same situation took place.
The problem appears at the comment `// Start tag:`  of htmlparser.js, specific code is `match = html.match( startTag );`.
Please see here [jsbin](http://jsbin.com/yikekig/1/edit?html,js,console,output), but please copy it to  someplace with a console.time

**Comment by alexlamsl** [comment]
Would you mind just pasting here an example HTML fragment which `html-minifier` chokes on?

I couldn't copy anything from your jsbin link because the tab just hangs there. And by "bug" do you mean a crash, an unexpected output, or something else?

**Comment by zyf0330** [comment]
This is 

``` js
var str = "objpicsurl_";
var unnormalAttr = 'ss"123';
var s = '<tag v-ref:vm_pv :imgs=" '+ str +' "></tag>';
var s1 = '  <tag v-ref:vm_pv :imgs=" '+ str +' " '+ unnormalAttr +'></tag>';
var s2 = '<tag v-ref:vm_pv :imgs=" '+ str +' " '+ unnormalAttr +'></tag>';
var startTag = /^<([\w:-]+)((?:\s*[\w:\.-]+(?:\s*(?:(?:=))\s*(?:(?:"[^"]*")|(?:'[^']*')|[^>\s]+))?)*)\s*(\/?)>/;

console.time('not exists unormal HTML tag');
s.match(startTag);
console.timeEnd('not exists unormal HTML tag');

console.log('exists unormal HTML tag and start with a whitespace');
for(var i in str){
    console.time(i);
    s1.replace(str, str.substring(0, str.length - i)).match(startTag);
    console.timeEnd(i);
}

console.log('exists unormal HTML tag and not start with a whitespace');
for(var i in str){
    console.time(i);
    s2.replace(str, str.substring(0, str.length - i)).match(startTag);
    console.timeEnd(i);
}
console.log('If I add a char into str, time increase to about double.');
```

**Comment by alexlamsl** [comment]
Can you paste the actual HTML content you'd pass to `html-minifier`?

**Comment by zyf0330** [comment]
Ok. `<tag v-ref:vm_pv :imgs=" objpicsurl_ " ss"123></tag>`, this is the bad HTML tag.

**Comment by zyf0330** [comment]
`console.log('If I add a char into str, time increase to about double.');`
This is the most important.

**Comment by alexlamsl** [comment]
It is invalid HTML/XML indeed. So I guess the problem here is you'd expect it to error out, but it is taking a long time before it'd do so?

**Comment by zyf0330** [comment]
Yeah, you are right. It is a invalid HTML tag. 
And regex match returns null as expect.
But it take so long time.

**Comment by zyf0330** [comment]
11 chars taste 4300ms, 12 chars taste 8400ms about double.
But it is not only one problem about invalid HTML tag.
`s1` is also an invalid HTML tag, but it starts with a tab, so regex match of it is so fast.
`s2` starts without a tab or other whitespaces, regex match of it is so slow, and time increases double with char length increass of attr value.

**Comment by zyf0330** [comment]
Sorry, I type "waste" wrongly to "taste".

**Comment by alexlamsl** [comment]
We are on the same page now :+1: 

I'll take a look at it - thanks for reporting!

**Comment by alexlamsl** [comment]
fix submitted in #510 - unfortunately it's on a branch other than `gh-pages`, so I can't give you an online version to test until it is merged :cry:

**Comment by zyf0330** [comment]
OK, will see later.  Thanks

On Thu, Feb 25, 2016, 18:44 Alex Lam S.L. notifications@github.com wrote:

> fix submitted in #510 https://github.com/kangax/html-minifier/pull/510
> - unfortunately it's on a branch other than gh-pages, so I can't give you
>   an online version to test until it is merged [image: :cry:]
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kangax/html-minifier/issues/507#issuecomment-188721840
> .
> 
> ## 

追随
